### ISSUE TITLE
[java] UnusedPrivateMethod: could not handle javax.annotation

### ISSUE THREAD

**Issue by Emilyaxe** [pmd/pmd#6627]
**Affects PMD Version:** 7.23.0

**Rule:** UnusedPrivateMethod — https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedprivatemethod

**Description:**

`UnusedPrivateMethod` incorrectly flags private methods annotated with `@javax.annotation.PostConstruct` and `@javax.annotation.PreDestroy` as unused. These methods are invoked reflectively by the container (e.g., CDI / Spring) during the application lifecycle and should therefore not be considered unused.

**Code Sample demonstrating the issue:**

```java
// (false-positive)
package demo.before;

import javax.annotation.PostConstruct;
import javax.annotation.PreDestroy;

public class LifecycleBean {
    private String message = "initialized";

    @PostConstruct
    private void setup() {              // UnusedPrivateMethod reported here
        this.message = "post-construct done";
    }

    @PreDestroy
    private void cleanup() {            // UnusedPrivateMethod reported here
        this.message = "pre-destroy done";
    }

    public String getMessage() { return message; }
}
```

**Expected outcome:** Private methods annotated with `@javax.annotation.PostConstruct` or `@javax.annotation.PreDestroy` should be treated as used (invoked reflectively by the container), and `UnusedPrivateMethod` should not report a violation.

**Actual outcome:** PMD reports `UnusedPrivateMethod` on methods annotated with `@javax.annotation.PostConstruct` and `@javax.annotation.PreDestroy`, resulting in a false positive.

**Running PMD through:** CLI

**Comment by UncleOwen** [comment]
Have you tried setting the `ignoredAnnotations` property?

**Comment by Emilyaxe** [comment]
> Have you tried setting the `ignoredAnnotations` property?

Thank you, I will carefully study the rule configuration.
